sdk/servicebus/azure_messaging_servicebus/src/receiver.rs

```diff
@@ -0,0 +1,3741 @@
+        let connection = create_test_connection();
+        let entity_name = "test-queue".to_string();
+        let options = create_test_options();
+
+        let receiver = Receiver::new(
+            connection,
+            entity_name,
+            None,
+            ReceiveMode::PeekLock,
+            options,
+        )
+        .await
+        .unwrap();
+
+        // Test with invalid "messages" field (not an array)
+        let mut response = azure_core_amqp::AmqpOrderedMap::new();
+        response.insert(
+            "messages".to_string(),
+            AmqpValue::String("invalid".to_string()),
+        );
+
+        let result = receiver.parse_peeked_messages_response(response).await;
+        assert!(result.is_err());
+        let error_msg = result.unwrap_err().to_string();
+        assert!(error_msg.contains("array"));
+    }
+
+    #[tokio::test]
+    async fn test_parse_peeked_messages_response_empty_array() {
+        let connection = create_test_connection();
+        let entity_name = "test-queue".to_string();
+        let options = create_test_options();
+
+        let receiver = Receiver::new(
+            connection,
+            entity_name,
+            None,
+            ReceiveMode::PeekLock,
+            options,
+        )
+        .await
+        .unwrap();
+
+        // Test with empty messages array
+        let mut response = azure_core_amqp::AmqpOrderedMap::new();
+        response.insert("messages".to_string(), AmqpValue::Array(vec![]));
+
+        let result = receiver.parse_peeked_messages_response(response).await;
+        assert!(result.is_ok());
+        let messages = result.unwrap();
+        assert_eq!(messages.len(), 0);
+    }
+
+    #[tokio::test]
+    async fn test_parse_peeked_messages_response_invalid_message_entry() {
+        let connection = create_test_connection();
+        let entity_name = "test-queue".to_string();
+        let options = create_test_options();
+
+        let receiver = Receiver::new(
+            connection,
+            entity_name,
+            None,
+            ReceiveMode::PeekLock,
+            options,
+        )
+        .await
+        .unwrap();
+
+        // Test with invalid message entry (not a map)
+        let mut response = azure_core_amqp::AmqpOrderedMap::new();
+        let messages_array = vec![AmqpValue::String("invalid".to_string())];
+        response.insert("messages".to_string(), AmqpValue::Array(messages_array));
+
+        let result = receiver.parse_peeked_messages_response(response).await;
+        assert!(result.is_err());
+        let error_msg = result.unwrap_err().to_string();
+        assert!(error_msg.contains("map"));
+    }
+
+    #[tokio::test]
+    async fn test_peek_messages_options_default() {
+        let options = PeekMessagesOptions::default();
+        assert_eq!(options.from_sequence_number, None);
+    }
+
+    #[tokio::test]
+    async fn test_peek_messages_options_with_sequence_number() {
```

Comment:
This test (a) doesn't seem to need to be async and (b) seems to be verifying that the compiler generates correct code.